Applications/Calendar/CalendarWidget.cpp

```diff
@@ -0,0 +1,177 @@
+                m_calendar_tiles[i] = add<CalendarTile>(*m_calendar, i, date_time);
+                m_calendar_tiles[i]->set_frame_thickness(0);
+                m_calendar_tiles[i]->set_relative_rect(x_offset, y_offset, 85, 85);
+            } else {
+                m_calendar_tiles[i]->update_values(*m_calendar, i, date_time);
+                m_calendar_tiles[i]->update();
+            }
+            i++;
+        }
+
+    m_calendar->set_selected_date(target_year, target_month);
+    m_selected_date_label->set_text(m_calendar->selected_date_text());
+}
+
+CalendarWidget::CalendarTile::CalendarTile(Calendar& calendar, int index, Core::DateTime date_time)
+    : m_index(index)
+    , m_date_time(date_time)
+    , m_calendar(calendar)
+{
+    update_values(calendar, index, date_time);
+}
+
+void CalendarWidget::CalendarTile::update_values(Calendar& calendar, int index, Core::DateTime date_time)
+{
+    m_calendar = calendar;
+    m_index = index;
+    m_date_time = date_time;
+    m_display_weekday_name = index < 7;
+
+    if (m_display_weekday_name) {
+        const String m_day_names[] = { "Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat" };
+        m_weekday_name = m_day_names[index];
+    }
+
+    m_display_date = (m_date_time.day() == 1) ? String::format("%s %d", name_of_month(m_date_time.month()).characters(), m_date_time.day()) : String::number(m_date_time.day());
+}
+
+CalendarWidget::CalendarTile::~CalendarTile()
+{
+}
+
+void CalendarWidget::CalendarTile::paint_event(GUI::PaintEvent& event)
+{
+    GUI::Frame::paint_event(event);
+
+    GUI::Painter painter(*this);
+    painter.fill_rect(frame_inner_rect(), Color::NamedColor::White);
+
+    painter.draw_line(frame_inner_rect().top_right(), frame_inner_rect().bottom_right(), Color::NamedColor::Black);
+    if (m_index == 0 || m_index % 7 == 0)
+        painter.draw_line(frame_inner_rect().top_left(), frame_inner_rect().bottom_left(), Color::NamedColor::Black);
+
+    if (m_index < 7)
+        painter.draw_line(frame_inner_rect().top_left(), frame_inner_rect().top_right(), Color::NamedColor::Black);
+    painter.draw_line(frame_inner_rect().bottom_left(), frame_inner_rect().bottom_right(), Color::NamedColor::Black);
+
+    Gfx::Rect day_rect;
+
+    if (m_display_weekday_name) {
+        auto weekday_rect = frame_inner_rect().shrunken(0, frame_inner_rect().height() / 1.2);
+        weekday_rect.set_top(frame_inner_rect().y() + 2);
+        painter.draw_text(weekday_rect, m_weekday_name, Gfx::Font::default_bold_font(), Gfx::TextAlignment::Center, Color::Black);
+
+        day_rect = frame_inner_rect().shrunken(0, frame_inner_rect().height() / 1.2);
+        day_rect.set_top(frame_inner_rect().y() + 15);
+    } else {
+        day_rect = frame_inner_rect().shrunken(0, frame_inner_rect().height() / 1.2);
+        day_rect.set_top(frame_inner_rect().y() + 4);
+    }
+
+    if (m_calendar.is_today(m_date_time.year(), m_date_time.month(), m_date_time.day())) {
```

Comment:
The usage of this API feels a little verbose. Maybe it would be better to just pass in m_date_time itself? Also, seems like a good candidate for a DateTime function IMO. Asking if it is today is quite expressive which other applications might be able to benefit from. We can const qualify this :)